src/ui/src/core_components/content/CoreAnnotatedText.vue

```diff
@@ -0,0 +1,339 @@
+	}
+
+	const colorStep = calculateColorStep(s, currentSteps.length);
+	const colorData = currentSteps[colorStep];
+
+	subjectColorCache[s] = colorData;
+	currentSteps.splice(colorStep, 1);
+
+	return generateColorCss(baseColor, colorData);
+}
+
+function copyText(arr: string[]) {
+	const text = arr.reduce((acc, val) => {
+		if (typeof val === "string") {
+			return acc + val;
+		}
+
+		return acc + val[0];
+	}, "");
+
+	copyToClipboard({ text });
+}
+
+function copyJSON(arr: string[]) {
+	try {
+		copyToClipboard({ text: JSON.stringify(arr) });
+	} catch (e) {
+		copyToClipboard({ text: arr.join("") });
+	}
+}
+
+function setClipboardData<T = unknown>(
+	source: T & { clipboardData: DataTransfer | null | undefined },
+	{ text, html }: { text?: string; html?: string },
+): void {
+	if (text) {
+		source.clipboardData?.setData("text/plain", text);
+		source.clipboardData?.setData("Text", text); // IE mimetype
+	}
+
+	if (html) {
+		source.clipboardData?.setData("text/html", html);
+	}
+}
+
+function copyToClipboard({
+	text = "",
+	html = "",
+}: {
+	text?: string;
+	html?: string;
+}): boolean {
+	if (
+		(window as any)?.clipboardData &&
+		(window as any)?.clipboardData.setData
+	) {
+		// Internet Explorer-specific code path to prevent textarea being shown while dialog is visible.
+		setClipboardData<any>(window, { text, html });
+
+		return true;
+	} else if (
+		document.queryCommandSupported &&
+		document.queryCommandSupported("copy")
+	) {
+		const copyListener = (event: ClipboardEvent) => {
+			event.preventDefault();
+			setClipboardData<ClipboardEvent>(event, { text, html });
+		};
+
+		document.addEventListener("copy", copyListener, false);
+
+		const textarea = document.createElement("textarea");
+		textarea.textContent = text || html;
+		textarea.style.position = "fixed";
+		document.body.appendChild(textarea);
+		textarea.select();
+
+		try {
+			return document.execCommand("copy"); // Security exception may be thrown by some browsers.
+		} catch (ex) {
+			return false;
+		} finally {
+			document.body.removeChild(textarea);
+			document.removeEventListener("copy", copyListener, false);
+		}
+	}
+
+	return false;
+}
+
+const defaultShortNameDictionary = {
+	verb: "Verb",
+	adjective: "Adj",
+	noun: "Noun",
+	pronoun: "Pronoun",
+};
+
+function getShortName(name: string) {
+	if (!name) {
+		return "";
+	}
+
+	const lowerCaseName = name.toLowerCase();
+	if (defaultShortNameDictionary[lowerCaseName]) {
+		return `${defaultShortNameDictionary[lowerCaseName]}`;
+	}
+
+	return name[0].toUpperCase() + name.slice(1);
+}
+</script>
+
+<style scoped>
+.AnnotatedText {
+	color: var(--primaryTextColor);
+	line-height: 1.8;
+}
+
+.annotation {
+	flex-direction: row;
+	align-items: center;
+	background: rgba(255, 164, 33, 0.4);
+	border-radius: 0.5rem;
+	padding: 0.08rem 0.5rem;
+	overflow: hidden;
+	line-height: 1;
+	vertical-align: middle;
+}
+
+.annotation-subject {
+	display: inline-flex;
+	font-size: 0.6rem;
```

Comment:
Let's use px instead of rem

I use rem for font-sizes, but looking to change to px too